### ISSUE TITLE
Feat: Connect device via WiFi

### ISSUE THREAD

**Issue by luisbytes** [okwasniewski/MiniSim#99]
It would be great if you could connect an Android v11+ device over WiFi by scanning the a QR Code just like Android Studio has it.

**Comment by gokul1099** [comment]
@okwasniewski  Shall I work on this

**Comment by okwasniewski** [comment]
Hey, thanks for opening a new feature request. MiniSim focuses on emulators and does not connect / display real devices. That's something worth considering in the future but firstly we would need to support connecting the device through USB. Also, the QR Code scanning might be exclusive to Android Studio and we would need to reimplement it. So let's keep this issue open but it could be worked on only after implementing USB devices support.

@gokul1099 If you want to you can work on adding support for USB devices (or take something else).

**Comment by gokul1099** [comment]
@okwasniewski  Ok, I will work on some other issue. Can you explain that USB devices support feature? or is there any separate issue raised for that??

**Comment by okwasniewski** [comment]
I will open a more detailed issue explaining what should be done to support this feature. For now let's focus on something else

**Comment by luisbytes** [comment]
Hi I don't have all the technical knowledge, but I think it is not an exclusive functionality of Android Studio. It generates a QR code this has an information that occupies ADB.

Here is an implementation with node
https://github.com/saleehk/adb-wifi/blob/master/src/index.js

**Comment by gokul1099** [comment]
> Hi I don't have all the technical knowledge, but I think it is not an exclusive functionality of Android Studio. It generates a QR code this has an information that occupies ADB.
> 
> Here is an implementation with node [saleehk/adb-wifi@`master`/src/index.js](https://github.com/saleehk/adb-wifi/blob/master/src/index.js?rgh-link-date=2023-11-23T14%3A17%3A16Z)

Yes, its a great plugin, I am using this for a long time, this works perfectly. I have also gone through implementation. It will be great if will add this to minisim.

**Comment by okwasniewski** [comment]
Thanks for sending! I wasn't aware of that in that case it looks like this feature can be added to MiniSIm

**Comment by gokul1099** [comment]
@okwasniewski  Shall i work on this ??

**Comment by okwasniewski** [comment]
@gokul1099 Sure, you can! However I think firstly it would be best to tackle #112 to finish that feature

**Comment by gokul1099** [comment]
> @gokul1099 Sure, you can! However I think firstly it would be best to tackle #112 to finish that feature

Ok let me try to fix that first

**Comment by okwasniewski** [comment]
Now that #122 has landed this can be revisited 👍🏻 

cc: @gokul1099
